hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/AbstractHoodieWriteClient.java

```diff
@@ -838,6 +860,36 @@ private HoodieTimeline getInflightTimelineExcludeCompactionAndClustering(HoodieT
     return inflightTimelineExcludeClusteringCommit;
   }
 
+  private Option<HoodiePendingRollbackInfo> getPendingRollback(HoodieTableMetaClient metaClient, String commitToRollback) {
+    Option<HoodiePendingRollbackInfo> pendingRollbackInfo = getPendingRollbacks(metaClient, Option.of(commitToRollback)).get(commitToRollback);
+    return pendingRollbackInfo != null ? pendingRollbackInfo : Option.empty();
+  }
+
+  /**
+   * Fetch map of pending commits to be rolledback to (Rollback Instant and Rollback plan).
+   * @param metaClient instance of {@link HoodieTableMetaClient} to use.
+   * @param commitToRollback optional commit time to be rolledback.
+   * @return map of pending commits to be rolledback instants to Rollback Instnat and Rollback plan Pair.
+   */
+  protected Map<String, Option<HoodiePendingRollbackInfo>> getPendingRollbacks(HoodieTableMetaClient metaClient, Option<String> commitToRollback) {
+    Stream<Pair<String, Option<HoodiePendingRollbackInfo>>> pendingRollbacksStream = metaClient.getActiveTimeline().filterPendingRollbackTimeline().getInstants().map(
+        entry -> {
+          try {
+            HoodieRollbackPlan rollbackPlan = RollbackUtils.getRollbackPlan(metaClient, entry);
+            return Pair.of(rollbackPlan.getInstantToRollback().getCommitTime(), Option.of(new HoodiePendingRollbackInfo(entry, rollbackPlan)));
+          } catch (IOException e) {
+            throw new HoodieIOException("Fetching rollback plan failed for " + entry, e);
+          }
+        }
+    );
+
+    if (commitToRollback.isPresent()) {
```

Comment:
could n't we move this to the call? keep this method clean - single purpose for each method